subsys/bluetooth/host/Kconfig

```diff
@@ -441,6 +441,7 @@ endif # BT_SMP
 
 rsource "Kconfig.l2cap"
 rsource "Kconfig.gatt"
+rsource "../services/Kconfig"
```

Comment:
If we are moving services to the host Kconfig, should we do the same for `mesh` and `audio`?

Reply:
That is a good point! I can do so if others agree

Reply:
I don't have the correct answer, but IMO `services` should be at the same level as `mesh` and `audio` given the subsystem structure. So if we move `services` from the common Kconfig to the host, then I think `mesh` and `audio` should too. 

That being said, `services` can also stay where it is, as that reflects the subsystem structure. 

@jhedberg Any input?

Reply:
AFAIK stuff like mesh, audio and services, even though logically layers above the host, were moved under `subsys/bluetooth` to avoid having super deep directory hierarchies. However yes, when it comes to Kconfig enablement, they should probably all depend on the host being enabled first and not reside somewhere higher up in the selection tree.

Reply:
Let's move the `audio` and `mesh` `rsource` lines to the same place as `services` then :)

Reply:
Adding audio resulted in some cyclic dependencies. Looking into it now

Reply:
I am not sure I see what the difference is between those Kconfigs now being rsourced from susbys/bluetooth/host/Kconfig, and those being rsourced from subsys/bluetooth/Kconfig

Should e.g. Kconfig.iso and Kconfig.adv also be rsourced here?
(Is adv by the way considered a part of the host?  It is in the spec.)

Reply:
I wanted to ensure those configs are not visible when performing a controller build. AFAIK the Kconfig.adv and Kconfig.iso contain global stack configuration options similar to `subsys/bluetooth/Kconfig`. Therefore, these cannot be guarded by the host.

Reply:
Maybe further refactoring is in place then - to split those into host, controller and common parts?

Also wondering the ways to do dependencies - rsourcing from within dependency as here, vs. select/depends.
Anyway - this PR is still an improvement to the current situation.

Reply:
Features that are common between the host and controller (e.g. adv and iso) makes sense to keep as common configs imo.